### DISCUSSION TITLE
How to import 'flushFailedPurchasesCachedAsPendingAndroid'?

### DISCUSSION THREAD

**Discussion by Simon-44** [hyochan/react-native-iap#2837] [Q&A]
Hello,

I'm migrating to 11 and the migration guide says that now, we must import platform-exclusive methods through 'IapAndroid', 'IapAmazon', 'IapIos' or 'IapIosSk2', giving the example of 'IapAndroid.validateReceiptAndroid' (https://react-native-iap.dooboolab.com/docs/migrate_to_11.0.0).

But the documentation about 'validateReceiptAndroid' still imports it from 'react-native-iap' (https://react-native-iap.dooboolab.com/docs/api-reference/methods/android/validate-receipt-android).

In my case, the only exclusive method seems to be 'flushFailedPurchasesCachedAsPendingAndroid', but if I try to import it through 'IapAndroid', I have a type error.

Thanks

**Comment by hyochan** [comment]
`flushFailedPurchasesCachedAsPendingAndroid` has been **completely removed** in react-native-iap v14+.

## Why?

The Google Play Billing library now handles failed purchases automatically with improved reconnection logic. Manual flushing is no longer necessary.

## Migration

### Before (v11/v12/v13)

```typescript
import { flushFailedPurchasesCachedAsPendingAndroid } from 'react-native-iap';

if (Platform.OS === 'android') {
  await flushFailedPurchasesCachedAsPendingAndroid();
}
```

### After (v14+)

```typescript
// Not needed - billing client handles this automatically
// Just use getAvailablePurchases() to restore pending purchases if needed

import { getAvailablePurchases } from 'react-native-iap';

const purchases = await getAvailablePurchases();
```

## Note

You mentioned migrating to v11, but v11/v12/v13 are **no longer supported**.

Please migrate directly to v14+:

- [Migration Guide (v13 to v14)](https://hyochan.github.io/react-native-iap/docs/guides/migration-v13-to-v14)
